File: VB/VB/VBClient/Form1.vb
Imports System
Imports Snap7

Public Class MainForm

    Dim Buffer(65536) As Byte
    Dim Client As Snap7.S7Client

    Private Sub ShowResult(ByVal Result As Integer)

        TextError.Text = Client.ErrorText(Result)
    End Sub

    Public Sub New()

        InitializeComponent()

        Client = New S7Client

        If IntPtr.Size = 4 Then
            Text = Text + " - Running 32 bit Code"
        Else
            Text = Text + " - Running 64 bit Code"
        End If

    End Sub

    Private Sub ConnectBtn_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ConnectBtn.Click
        Dim Result As Integer
        Dim Rack As Integer = System.Convert.ToInt32(TxtRack.Text)
        Dim Slot As Integer = System.Convert.ToInt32(TxtSlot.Text)
        Result = Client.ConnectTo(TxtIP.Text, Rack, Slot)
        ShowResult(Result)
        If Result = 0 Then
            TxtIP.Enabled = False
            TxtRack.Enabled = False
            TxtSlot.Enabled = False
            ConnectBtn.Enabled = False
            DisconnectBtn.Enabled = True
            TxtMB.Enabled = True
            TxtSize.Enabled = True
            RBtn.Enabled = True
            RBox.Enabled = True
            WBtn.Enabled = True
            WBox.Enabled = True
        End If
    End Sub

    Private Sub DisconnectBtn_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles DisconnectBtn.Click
        Client.Disconnect()
        TxtIP.Enabled = True
        TxtRack.Enabled = True
        TxtSlot.Enabled = True
        ConnectBtn.Enabled = True
        DisconnectBtn.Enabled = False
        TxtMB.Enabled = False
        TxtSize.Enabled = False
        RBtn.Enabled = False
        RBox.Enabled = False
        WBtn.Enabled = False
        WBox.Enabled = False
    End Sub

    Private Sub RBtn_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles RBtn.Click

        Dim MB As Integer
        Dim Size As Integer
        Dim Result As Integer
        Dim c, y As Integer
        Dim s As String
        '
        RBox.Text = ""
        MB = System.Convert.ToInt32(TxtMB.Text)
        Size = System.Convert.ToInt32(TxtSize.Text)

        Result = Client.MBRead(0, Size, Buffer)
        ShowResult(Result)

        If Result = 0 Then
            y = 0
            For c = 0 To Size - 1
                s = Hex$(Buffer(c))
                If s.Length = 1 Then
                    s = "0" + s
                End If
                RBox.Text = RBox.Text + "0x" + s + " "
                y = y + 1
                If y = 8 Then
                    y = 0
                    RBox.Text = RBox.Text + Chr(13) + Chr(10)
                End If
            Next
        End If
    End Sub
    Private Sub WBtn_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles WBtn.Click

        Dim MB As Integer
        Dim Size1 As Integer
        Dim Result1 As Integer
        Dim c1, y1 As Integer
        Dim s1 As String
        '
        WBox.Text = ""
        MB = System.Convert.ToInt32(TxtMB.Text)
        Size1 = System.Convert.ToInt32(TxtSize.Text)

        Result1 = Client.MBWrite(0, Size1, Buffer)
        ShowResult(Result1)

        If Result1 = 0 Then
            y1 = 0
            For c1 = 0 To Size1 - 1
                s1 = Hex$(Buffer(c1))
                If s1.Length = 1 Then
                    s1 = "0" + s1
                End If
                WBox.Text = WBox.Text + "0x" + s1 + " "
                y1 = y1 + 1
                If y1 = 8 Then
                    y1 = 0
                    WBox.Text = WBox.Text + Chr(13) + Chr(10)
                End If
            Next
        End If
    End Sub
    Private Sub TxtDB_TextChanged(sender As Object, e As EventArgs) Handles TxtMB.TextChanged

    End Sub

    Private Sub MainForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub
End Class

File: Lane_Simulator_DEMO/Form1.cs
using System;
using System.Runtime.InteropServices;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Snap7;

namespace CSClient
{

    public partial class MainForm : Form
    {
        private S7Client Client;
        private byte[] Buffer = new byte[65536];
        
        //Inputs
        public static int Start_Button = 95;
        public static int Stop_Button = 94;
       
         // outputs 
        public static int StartButton = 95;
        public static int StopButton = 94;
        
        public static int System_Reset = 93;
        // Merkers
        public static int System_Start = 7336;
        public static int System_Stop = 7337;


        private void ShowResult(int Result)
        {
            // This function returns a textual explaination of the error code
            TextError.Text = Client.ErrorText(Result);
        }

        public MainForm()
        {
            InitializeComponent();
            Client = new S7Client();
            if (IntPtr.Size == 4)
                this.Text = this.Text + " - Running 32 bit Code";
            else
                this.Text = this.Text + " - Running 64 bit Code";

            //CBType.SelectedIndex = 0;
        }

        private void ConnectBtn_Click(object sender, EventArgs e)
        {
            int Result;
            int Rack = System.Convert.ToInt32(TxtRack.Text);
            int Slot = System.Convert.ToInt32(TxtSlot.Text);
            Result = Client.ConnectTo(TxtIP.Text, Rack, Slot);
            ShowResult(Result);
            
            if (Result == 0)
            {
                
                // Make connection with PLC using IP, slot, etc..
                TxtIP.Enabled = false;
                TxtRack.Enabled = false;
                TxtSlot.Enabled = false;
                ConnectBtn.Enabled = false;
                DisconnectBtn.Enabled = true;

                AddressField.Enabled = true;
                ValueField.Enabled = false;
              
                WriteBtn.Enabled = true;
                ReadBtn.Enabled = true;

                ComboBox1.Enabled = true;

            }
        }

        private void DisconnectBtn_Click(object sender, EventArgs e)
        {
            Client.Disconnect();
            TxtIP.Enabled = true;
            TxtRack.Enabled = true;
            TxtSlot.Enabled = true;
            ConnectBtn.Enabled = true;
            DisconnectBtn.Enabled = false;

            AddressField.Enabled = false;
            ValueField.Enabled = false;
           
            WriteBtn.Enabled = false;
            ReadBtn.Enabled = false;

            ComboBox1.Enabled = false;

        }




       

        private void WriteBtn_Click(object sender, EventArgs e)
        {
            if (ComboBox1.SelectedItem == "Inputs")
            {
                InputWrite();
            }
            if (ComboBox1.SelectedItem == "Outputs")
            {
                OutputWrite();
            }
        }

        private void ReadBtn_Click(object sender, EventArgs e)
        {
            if (ComboBox1.SelectedItem == "Inputs")
            {
                ValueField.Enabled = true;
                InputRead();
            }
            if (ComboBox1.SelectedItem == "Outputs")
            {
                OutputRead();
            }
        }

 
        private void InputRead()
        {
            // Declaration separated from the code for readability
            int DBNumber = 0;
            int start;
            int amount = 1;
            byte[] buffer = new byte[1];

            int area = S7Client.S7AreaPE;
            int Result;
            start = System.Convert.ToInt32(AddressField.Text);

            Result = Client.ReadArea(area, DBNumber, start, amount, S7Client.S7WLBit, buffer);

            if (Result == 0)
            {
                if (start == Start_Button)
                {
                    if (buffer[0] == 1)
                    {
                        Start_Button_Panel.BackColor = Color.Green;
                    }

                    else
                    {
                        Start_Button_Panel.BackColor = Color.DarkGray;
                    }
                }
                if (start == Stop_Button)
                {
                    if (buffer[0] == 1)
                    {
                        Stop_Button_Panel.BackColor = Color.Green;
                    }

                    else
                    {
                        Stop_Button_Panel.BackColor = Color.DarkGray;
                    }
                }

                if (start == System_Reset)
                {
                    if (buffer[0] == 1)
                    {
                        System_Button_Panel.BackColor = Color.Green;
                    }

                    else
                    {
                        System_Button_Panel.BackColor = Color.DarkGray;
                    }
                }

            }

        }


        private void InputWrite()
        {
            int DBNumber = 0;
            int amount = 1;
            int start;
            int area = S7Client.S7AreaPE;
            int Result;
            start = System.Convert.ToInt32(AddressField.Text);

            byte[] buffer = new byte[1];

            Byte.TryParse(ValueField.Text, out buffer[0]);

            Result = Client.WriteArea(S7Client.S7AreaPE, DBNumber, start, amount, S7Client.S7WLBit, buffer);
            
            if (Result == 0)
            {
                if (start == Start_Button)
                {
                    if (buffer[0] == 1)
                    {
                        Start_Button_Panel.BackColor = Color.Green;
                    }

                    else
                    {
                        Start_Button_Panel.BackColor = Color.DarkGray;
                    }
                }
                if (start == Stop_Button)
                {
                    if (buffer[0] == 1)
                    {
                        Stop_Button_Panel.BackColor = Color.Green;
                    }

                    else
                    {
                        Stop_Button_Panel.BackColor = Color.DarkGray;
                    }
                }

                if (start == System_Reset)
                {
                    if (buffer[0] == 1)
                    {
                        System_Button_Panel.BackColor = Color.Green;
                    }

                    else
                    {
                        System_Button_Panel.BackColor = Color.DarkGray;
                    }
                }
                
            }
      
        }



        private void OutputRead()
        {
            int DBNumber = 0;
            int amount = 1;
            int start;
            int area = S7Client.S7AreaPA;
            int Result;

            start = System.Convert.ToInt32(AddressField.Text);

            byte[] buffer = new byte[1];
            
            Byte.TryParse(ValueField.Text, out buffer[0]);

            Result = Client.ReadArea(area, DBNumber, start, amount, S7Client.S7WLBit, buffer);
        }


        private void OutputWrite()
        {
            // Declaration separated from the code for readability
            int DBNumber = 0;
            int amount = 1;
            int start;
            int area = S7Client.S7AreaPA;
            int Result;
            start = System.Convert.ToInt32(AddressField.Text);

            byte[] buffer = new byte[1];

            Byte.TryParse(ValueField.Text, out buffer[0]);
            
        
            Result = Client.WriteArea(area, DBNumber, start, amount, S7Client.S7WLBit, buffer);

            if (Result == 0)
            {
                if (start == StartButton)
                {
                    if (buffer[0] == 1)
                    {
                        StartButtonPanel.BackColor = Color.Green;
                    }

                    else
                    {
                        StartButtonPanel.BackColor = Color.DarkGray;
                    }
                }
                if (start == StopButton)
                {
                    if (buffer[0] == 1)
                    {
                        StopButtonPanel.BackColor = Color.Green;
                    }

                    else
                    {
                        StopButtonPanel.BackColor = Color.DarkGray;
                    }
                }
            }
            
            OutputRead();
      
        }



        private void realTimeSimulation()
        {
            int DBNumber = 0;
            int amount = 1;
            int start;
            int area = S7Client.S7AreaPA;
            int Result;
            start = System.Convert.ToInt32(AddressField.Text);

            byte[] buffer = new byte[1];

            Byte.TryParse(ValueField.Text, out buffer[0]);

            Result = Client.ReadArea(area, DBNumber, start, amount, S7Client.S7WLBit, buffer);

            if (Result == 0)
            {
      
            }
        }


  

    }
}


